src/Features/RoyalShaving/components/StageHeader/StageHeader.tsx

```diff
@@ -0,0 +1,82 @@
+import { ReactElement, PropsWithChildren } from 'react';
+
+type HeaderPositionVariant = 'headerLeft' | 'headerRight';
+type TitlePositionVariant = 'titleLeft' | 'titleRight';
+type ImagePositionVariant = 'imageLeft' | 'imageRight';
+
+interface StageHeaderProps extends PropsWithChildren {
+  headerTitleTop: string;
+  headerTitleBottom: string;
+  headerPosition: HeaderPositionVariant;
+  titlePosition: TitlePositionVariant;
+  imagePosition: ImagePositionVariant;
+}
+
+export const StageHeader = ({
+  headerTitleTop,
+  headerTitleBottom,
+  headerPosition,
+  titlePosition,
+  imagePosition,
+  children,
+}: StageHeaderProps): ReactElement => {
+  const classes = {
```

Comment:
you can improve it a bit from the type safety perspective:

type PositionVariant = HeaderPositionVariant | TitlePositionVariant | ImagePositionVariant;

const classes: { [K in PositionVariant]: string }= {
    headerLeft: 'stage__header mr-10',
    headerRight: 'stage__heade_reversed ml-10 mr-0',
    titleLeft: 'after:-right-5',
    titleRight: 'mr-0 ml-auto after:right-auto after:-left-5 after:rotate-180',
    imageLeft: 'stage__image after:-right-5',
    imageRight: 'stage__image_reversed -left-10 after:right-auto after:left-0 after:rotate-180',
};


now, this object will cause errors if some properties are not implemented

Reply:
Done